### PR TITLE
RESTWS-796: Local variables should not be declared and then immediate…

### PR DESCRIPTION
**Pull request by kdaud**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-796
## Checklist: I completed these to help reviewers :)

- [ X] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java [hunk 1]
```diff
@@ -526,8 +526,7 @@ public Object upload(MultipartFile file, RequestContext context) throws Response
 		
 		obs = obsService.saveObs(obs, null);
 		
-		SimpleObject ret = (SimpleObject) ConversionUtil.convertToRepresentation(obs, Representation.DEFAULT);
-		return ret;
+		return ConversionUtil.convertToRepresentation(obs, Representation.DEFAULT);
```
Comment (HerbertYiga) [inline omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java:529]: Does this line return SimpleObject   objtect type without casting?

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java]: Have updated the code base

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java]: Have updated the code base

> Does this line return SimpleObject objtect type without casting?

Have updated the code base

Comment (kdaud) [reply omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java]: cc: @sacull

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/DelegatingCrudResource.java [hunk 2]
```diff
@@ -175,8 +175,7 @@ public Object undelete(String uuid, RequestContext context) throws ResponseExcep
 			throw new ObjectNotFoundException();
 		
 		delegate = undelete(delegate, context);
-		SimpleObject ret = (SimpleObject) ConversionUtil.convertToRepresentation(delegate, context.getRepresentation());
-		return ret;
+		return ConversionUtil.convertToRepresentation(delegate, context.getRepresentation());
```
Comment (sacull) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/DelegatingCrudResource.java:178]: Shouldn't you first cast to a SimpleObject here?

File: omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImpl.java [hunk 3]
```diff
@@ -661,6 +659,7 @@ private void eliminateCandidateSearchHandlersWithMissingRequiredParameters(Set<S
 	 * <strong>Should</strong> return all search handlers if search handlers have been initialized
 	 * <strong>Should</strong> return null if search handlers have not been initialized
 	 */
+	@Override
```
Comment (sacull) [inline omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImpl.java:662]: There are other tickets for this activity. Yours doesn't include it. ;)

File: omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/SystemSettingResource1_9.java [hunk 4]
```diff
@@ -11,9 +11,6 @@
 
 import java.util.List;
 
-import io.swagger.models.Model;
-import io.swagger.models.ModelImpl;
-import io.swagger.models.properties.StringProperty;
```
Comment (sacull) [inline omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/SystemSettingResource1_9.java:16]: By changing the setting of the imports you make someone who is viewing the code wondering why this is so red. I guess it's your IDE's doing?

File: omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java [hunk 5]
```diff
@@ -9,12 +9,9 @@
...
 
-import io.swagger.models.Model;
-import io.swagger.models.ModelImpl;
-import io.swagger.models.properties.BooleanProperty;
-import io.swagger.models.properties.ObjectProperty;
-import io.swagger.models.properties.RefProperty;
-import io.swagger.models.properties.StringProperty;
+import java.util.ArrayList;
+import java.util.List;
+
```
Comment (sacull) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderSetMemberResource1_12.java:14]: Same as above.

**Comment by kdaud** [comment]
Have updated the PR.
cc: @sacull

**Comment by sacull** [comment]
image

File: omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java [hunk 6]
```diff
@@ -155,18 +155,16 @@ public Model getGETModel(Representation rep) {
...
 		        .property("orders", new ArrayProperty(new RefProperty("#/definitions/OrderCreate")))
 		        .property("orderSet", new StringProperty().example("uuid"));
-		return model;
 	}
 	
 	@Override
 	public Model getUPDATEModel(Representation rep) {
-		ModelImpl model = new ModelImpl().property("orders",
+		return new ModelImpl().property("orders",
 		    new ArrayProperty(new RefProperty("#/definitions/OrderCreate")));
```
Comment (sacull) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]:
```suggestion
		return new ModelImpl().property("orders", new ArrayProperty(new RefProperty("#/definitions/OrderCreate")));
```
Why not in one line?

Comment (sacull) [inline omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]:
```suggestion
		return new ModelImpl().property("orders", new ArrayProperty(new RefProperty("#/definitions/OrderCreate")));
```
Why not in one line?

Comment (kdaud) [reply omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]: Hey @sacull 
Setting the IDE as pre-described by this wiikik https://wiki.openmrs.org/display/docs/Developer+How-To+Setup+And+Use+Eclipse does automatically  code formatting on save action. This action formats the source codes as per OpenMRS standards!

Comment (sacull) [reply omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]: It looks like you might be right. I was based on: 
> Long lines. Lines should not exceed 125 characters.
source: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Others

> <setting id="org.eclipse.jdt.core.formatter.comment.line_length" value="100"/>
source: https://github.com/openmrs/openmrs-core/blob/master/tools/src/main/resources/eclipse/OpenMRSFormatter.xml#L200

So one document says 125 characters, another 100 characters, so we can assume that you are right and we can leave your version. ;)

Comment (kdaud) [reply omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]: Is there any other changes that I need to address concerning this ticket?
cc: @sacull

Comment (sacull) [reply omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]: https://github.com/openmrs/openmrs-module-webservices.rest/pull/472#issuecomment-728675452 ;)

Comment (kdaud) [reply omod-1.12/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_12/OrderGroupResource1_12.java:167]: Hey @sacull  have addressed the comment as per say

**Review by sacull — APPROVED** [review]
Well, it looks good to me now.

**Comment by dkayiwa** [comment]
Thanks @kdaud for working on this and thanks @sacull for reviewing. 😊 

@sacull i have given you Write access to this repository. So feel free to merge this pull request and the next ones that you will review and approve. I have also given you permission to the corresponding JIRA project such that you are in position to close these tickets. 👍

**Comment by sacull** [comment]
@dkayiwa Oh, thank you, it's a big responsibility. I'll try not to break anything.

**Comment by dkayiwa** [comment]
You have been responsible enough to trust you with that! 😊

**Pull request merged**